specs/casper_sharding_v2.1.md

```diff
@@ -581,27 +581,43 @@ Finally:
 * Let `next_start_shard = (shard_and_committee_for_slots[-1][-1].shard_id + 1) % SHARD_COUNT`
 * Set `shard_and_committee_for_slots[CYCLE_LENGTH:] = get_new_shuffling(block.ancestor_hashes[0], validators, next_start_shard)`
 
--------
-
-Note: this is ~80% complete. The main sections that are missing are:
-
-* Logic for the formats of shard chains, who proposes shard blocks, etc. (in an initial release, if desired we could make crosslinks just be Merkle roots of blobs of data; in any case, one can philosophically view the whole point of the shard chains as being a coordination device for choosing what blobs of data to propose as crosslinks)
-* Logic for inducting queued validators from the PoW chain
-* Penalties for signing or attesting to non-canonical-chain blocks (update: may not be necessary, see https://ethresear.ch/t/attestation-committee-based-full-pos-chains/2259)
-* Per-validator proofs of custody, and associated slashing conditions
-* Versioning and upgrades
-
-Slashing conditions may include:
-
-
-    Casper FFG slot equivocation [done]
-    Casper FFG surround [done]
-    Beacon chain proposal equivocation [done]
-    Shard chain proposal equivocation
-    Proof of custody secret leak
-    Proof of custody wrong custody bit
-    Proof of custody no secret reveal
-    RANDAO leak
+### TODO
+
+Note: This spec is ~60% complete.
+
+* [ ] Specify how `crystallized_state_root` and `active_state_root` are constructed
+* [ ] Specify the shard chain blocks, blobs, proposers, etc.
+* [ ] Specify the rules for forced deregistrations
+* [ ] Add a double-batched Merkle accumulator for historical beacon chain blocks
+* [ ] Specify the various assumptions (global clock, validator honesty, validator liveness, etc.)
+* [ ] Specify in Vyper the registration contract on the PoW chain
+* [ ] Specify the RANDAO logic, including the hardening against orphaned reveals
+* [ ] Add per-validator proofs of custody, including slashing conditions
+* [ ] Use a separate hash function for the proof of possession
+* [ ] Add an appendix for BLS12-381
+* [ ] Add an appendix on gossip networks and the offchain signature aggregation logic
+* [ ] Comprehensively and clearly define all the terms in the glossary
+* [ ] Clearly document the various edge cases, e.g. with committee sizing
+* [ ] Rework the document for readability
+* [ ] Undergo peer review, security audits and formal verification
+
+**Possible changes and additions**
+
+* [ ] Deprecate Wei and use 64-bit balances, consistent with a 64-bit EVM2.0
```

Comment:
eWASM is planning to use `uint128` for eth balances for the full Wei range.
Turns out that pretty much all modern architectures have some built in support for handling int128 pretty efficiently.

 I think we can remove this and only add _if_ eWASM changes roadmap on this

Reply:
If all modern architectures support int128 does it mean that the native word size for eWASM is 128 bits (as opposed 64 bits)?

Reply:
architectures support some tricks to do pretty optimized 128 bit arithmetic. The native register size is still 64 and the native word size of eWASM is 64.

I reached out to the eWASM team regarding this issue a couple of weeks ago and this was what they told me.